client/src/components/DeleteButton.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import styled from 'styled-components';
+
+const DeleteButton = ({ name, id, isSubscribed, setIsSubscribed }) => {
+  const HOST = 'http://127.0.0.1:5000/';
+
+  const handleDeleteRequest = async () => {
+    try {
+      const resp = await fetch(`${HOST}/watchlist`, {
+        method: 'DELETE',
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ id }),
+      }).then((res) => {
+        console.log(res);
+      });
+    } catch (e) {
+      console.log(e, 'Failed to delete channel');
+    }
+    setIsSubscribed(false);
```

Comment:
Also need to check if `resp.ok` is true and only then we should call the callback from the parent. Otherwise, it will be called even if the request isn't successful.

Reply:
I think I understood what you meant. I added an if-statement inside the try block to check if `resp.ok` is true and call `setIsSubscribed` if it is.